### PR TITLE
Document and use the new atomistic models naming convention

### PR DESCRIPTION
**Pull request by Luthaf**
`System` has been renamed to `Structure` and `species` has been renamed to `types`. This mainly affects metatensor-torch, but I also updated examples/tests names to follow the new convention. I'll still have to update the example data once we reach a decision on all the other names.

# Contributor (creator of pull-request) checklist

 - [x] Tests updated (for new features and bugfixes)?
 - [x] Documentation updated (for new features)?
 - [ ] ~Issue referenced (for PRs that solve an issue)?~

# Reviewer checklist

 - [ ] CHANGELOG updated with public API or any other important changes?

----
📚 Documentation preview 📚: https://metatensor--514.org.readthedocs.build/en/514/

### CODE REVIEW

File: docs/src/atomistic/outputs.rst [hunk 1]
```diff
@@ -16,20 +16,105 @@ to this page.
...
+
+Energy is associated with the ``"energy"`` key in the model outputs, and must
+have the following metadata:
+
+.. list-table:: Metadata for energy output
+  :widths: 2 3 7
+  :header-rows: 1
+
+  * - Metadata
+    - Names
```
Comment (frostedoyster) [inline docs/src/atomistic/outputs.rst:29]: Names? Of who/what? Maybe name of the energy? Like U0?

File: docs/src/core/overview.rst [hunk 2]
```diff
@@ -47,7 +47,7 @@ dimensions you'll encounter when working with atomistic data are the following:
   harmonics. When handling this kind of data, it is convenient to store and
   manipulate the data corresponding to different spherical harmonics (or
   generally different irreducible representations of the symmetry group)
-  separately. This is the case of the ``spherical_harmonics_l`` key dimension
+  separately. This is the case of the ``o3_lambda`` key dimension
   produced by `rascaline`_: different blocks will contain the :math:`\ell = 1`
   and :math:`\ell = 2` parts of an equivariant representation.
```
Comment (frostedoyster) [inline docs/src/core/overview.rst:52]: lambda?

File: docs/src/atomistic/outputs.rst [hunk 3]
```diff
@@ -16,20 +16,105 @@ to this page.
...
+      the equation above.
+
+  * - components
+    - ``"xyz"``
+    - there must be a single component named ``"xyz"`` with values 0, 1, 2;
+      indicating the direction of the displacement of the atom in the gradient
+      samples.
+
+- **"strain"** (:math:`\epsilon`) gradients will contain the stress
+  :math:`\sigma` acting on the system, scaled by the volume :math:`V`
```
Comment (frostedoyster) [inline docs/src/atomistic/outputs.rst:98]: Perhaps also mention the virial?

File: docs/src/atomistic/outputs.rst (hunk 1 above)
Comment (Luthaf) [reply docs/src/atomistic/outputs.rst:29]: This is (one of) the syntax for tables in rst. Have you looked at how this is rendered? 

https://metatensor--514.org.readthedocs.build/en/514/atomistic/outputs.html

**Review by frostedoyster — APPROVED** [review]

**Pull request merged**